packages/react-native/ReactCommon/react/renderer/textlayoutmanager/platform/ios/react/renderer/textlayoutmanager/RCTTextLayoutManager.mm

```diff
@@ -152,15 +152,17 @@ - (LinesMeasurements)getLinesForAttributedString:(facebook::react::AttributedStr
                                                      attribute:NSFontAttributeName
                                                        atIndex:0
                                                 effectiveRange:nil];
+                                            CGRect lineRect = [layoutManager lineFragmentRectForGlyphAtIndex:range.location effectiveRange:nil];
+                                            CGFloat baseline = [layoutManager locationForGlyphAtIndex:range.location].y;
```

Comment:
I suspect this might not always be right, if glyphs can have parts drawn below the baseline. E.g. the descender of lowercase "g" is under the baseline.

I wonder if we could get the right answer if we used the font for the right glyphs. Though I think we'd also need to incorporate any offsets to baseline that we apply.

Reply:
> I suspect this might not always be right, if glyphs can have parts drawn below the baseline. E.g. the descender of lowercase "g" is under the baseline.

I'm not sure what you mean. Isn't the y-coordinate of the point returned by `lineFragmentRectForGlyphAtIndex` a distance between the top of the line rect and the baseline of the glyph?

> I wonder if we could get the right answer if we used the font for the right glyphs. Though I think we'd also need to incorporate any offsets to baseline that we apply.

I think it's possible - we would need to iterate over all fonts and use the largest ascender and descender as the values for the line measurements, no? It would also need to take into account the height of the attachments, in case they are higher than the largest ascender. Not sure about the baseline offsets, quick and dirty test didn't really show any difference between that approach and the one currently implemented. I think I would need to look more closely into this if we want to go in this direction.

Reply:
Where do you see that it would be the baseline? I was looking for documentation suggesting something like this, but couldn’t find it. 

The problem would be, if the API returned the bounding box of the glyph, since the bottom of the glyph may go under (or above) the baseline.

Reply:
> Where do you see that y coordinate of locationForGlyphAtIndex would be the baseline? I was looking for documentation suggesting something like this, but couldn’t find it.

I didn't find it in the documentation either, but to me only three values are sensible here:
- distance to the top of the glyph
- distance to the bottom of the glyph
- distance to the baseline of the glyph

And looking at the values it's returning, it's not the first two - the value seems to be constant for all glyphs in the line and is too large to be one of them (for example 24 in a line with a height of 30). It also seems to give the correct results:

<img width="519" alt="Screenshot 2024-07-08 at 17 21 51" src="https://github.com/facebook/react-native/assets/21055725/f017059c-ccf4-466e-a63b-ff8626a9da40">

I can try to find another way of finding the baseline if this isn't acceptable.

Reply:
If this is always returning the baseline, I think we should be okay. Thank you for checking!